### ISSUE TITLE
Several crashes / errors caused by numeric filenames

### ISSUE THREAD

**Issue by gerben-van-eck** [sabnzbd/sabnzbd#1575]
I use the NZB Unity browser extension, this sends an nzb to sabnzbd through the API. The filename is set to a random number, this seems to get treated as an int instead of a numeric string, which causes some errors, namely:

```
2020-08-15 09:52:02,675::INFO::[postproc:367] Starting Post-Processing on 293894103 => Repair:True, Unpack:True, Delete:True, Script:None, Cat:*
2020-08-15 09:52:02,680::INFO::[postproc:722] Starting verification and repair of 293894103
2020-08-15 09:52:02,708::INFO::[postproc:794] Verification and repair finished for 293894103
2020-08-15 09:52:02,709::ERROR::[postproc:600] Post Processing Failed for 293894103 (see logfile)
2020-08-15 09:52:02,714::INFO::[postproc:601] Traceback: 
Traceback (most recent call last):
  File "/opt/sabnzbd/sabnzbd/postproc.py", line 415, in process_job
    tmp_workdir_complete, workdir_complete, file_sorter, one_folder, marker_file = prepare_extraction_path(
  File "/opt/sabnzbd/sabnzbd/postproc.py", line 693, in prepare_extraction_path
    workdir_complete = get_unique_path(os.path.join(complete_dir, nzo.final_name), create_dir=True)
  File "/usr/lib64/python3.8/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib64/python3.8/genericpath.py", line 152, in _check_arg_types
    raise TypeError(f'{funcname}() argument must be str, bytes, or '
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'int'
```
In sorting.py I already hacked a try/except block in the check_regexs method, the expression.search fails for the same reason.
My Python knowledge is limited, but I guess a string casting should be added somewhere.

**Comment by Safihre** [comment]
Could you maybe also share the rest of your settings? If you click Show Logging in the Status window, it has a anonymous copy of your settings at the end.

**Comment by Safihre** [comment]
I think I found the cause. Just to double check: do you use "Check before download" option?

**Comment by gerben-van-eck** [comment]
[settings.txt](https://github.com/sabnzbd/sabnzbd/files/5080390/settings.txt)

These are my settings.

I could not find the 'check before download' option.

**Comment by gerben-van-eck** [comment]
I've added a last hack to make it 'work for me'. This will not likely be a neat final solution, but I hope with knowledge of the codebase you know where to fix this properly.

`diff --git a/sabnzbd/postproc.py b/sabnzbd/postproc.py
index 98bdf598..dada1daa 100644
--- a/sabnzbd/postproc.py
+++ b/sabnzbd/postproc.py
@@ -687,6 +687,7 @@ def prepare_extraction_path(nzo):
 
     complete_dir = sanitize_and_trim_path(complete_dir)
 
+    nzo.final_name = str(nzo.final_name)
     if one_folder:
         workdir_complete = create_all_dirs(complete_dir, umask=True)
     else:`

**Comment by Safihre** [comment]
Does it always happen or when you Retry a job?

**Comment by gerben-van-eck** [comment]
When I retry a job. Most dl's are encrypted so I start these, they fail, enter password, retry. Now I think of it the jobs that are not encrypted go well.

**Comment by Safihre** [comment]
Found the problem, will be fixed in 3.1.0.

**Comment by Safihre** [comment]
Fix for the 3.0.x series would be to replace these 2 lines: https://github.com/sabnzbd/sabnzbd/commit/b063055e78c6c3199f12efb3d330c8081afe9f5a#diff-f13b6c9185bd30609eccd1f3572729ccL778-L779
With:
```python
            if name:
                self.final_name = str(name)
```
